src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -125,6 +125,7 @@ private void AddSource(string fileName, string source, bool isRootContextDef = f
                 StringBuilder sb = new();
 
                 sb.Append($@"// <auto-generated/>
+#nullable enable
```

Comment:
Enable for entire project? If not, do you want a `#nullable restore` later.

Reply:
Don't think we can (or want to) enable `nullable` for the consuming project from the generator. The goal is not to cause warnings in projects with nullable enabled.

I believe the directive affects only each generated file, so we shouldn't need to restore at the end. cc @maryamariyan @eerhardt

Reply:
Yes, it's scoped to the particular file.